flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -154,8 +160,12 @@ public RocksIncrementalSnapshotStrategy(
 		final SnapshotDirectory snapshotDirectory = prepareLocalSnapshotDirectory(checkpointId);
 		LOG.trace("Local RocksDB checkpoint goes to backup path {}.", snapshotDirectory);
 
+		if (this.checkpointStreamFactory == null) {
+			this.checkpointStreamFactory = checkpointStreamFactory;
```

Comment:
The line 175 may use this.checkpointStreamFactory replace  checkpointStreamFactory.